### PR TITLE
Anakin stop panicking

### PR DESCRIPTION
**Pull request by knoellle**
## Introduced Changes

Upon panic, always prints location regardless of payload.
On main, only the payload of `panic!("hello world")` is printed, but not the one of `panic!("{some_variable}")`.

We only checked for `&str`, but panic with format string has a `String` payload.

Fixes #

## How to Test

Insert panics somehwere and observe how they are printed.

### CODE REVIEW

File: crates/code_generation/src/execution.rs [hunk 1]
```diff
@@ -39,10 +39,16 @@ pub fn generate_run_function(cyclers: &Cyclers) -> TokenStream {
...
+                    eprint!("panic occurred");
                     if let Some(payload) = panic_info.payload().downcast_ref::<&str>() {
-                        eprintln!("panic occurred: {payload:?} {:?}", panic_info.location());
-                    } else {
-                        eprintln!("panic occurred");
+                        eprint!(": {payload:?}");
+                    }
+                    if let Some(payload) = panic_info.payload().downcast_ref::<String>() {
+                        eprint!(": {payload:?}");
+                    }
```
Comment (schmidma) [inline crates/code_generation/src/execution.rs:48]: Can we cast to something like `Box<dyn Debug>`?

File: crates/code_generation/src/execution.rs [hunk 2]
```diff
@@ -39,10 +39,16 @@ pub fn generate_run_function(cyclers: &Cyclers) -> TokenStream {
...
+                    eprintln!();
```
Comment (schmidma) [inline crates/code_generation/src/execution.rs:49]: why?

File: crates/code_generation/src/execution.rs (hunk 1 above)
Comment (schmidma) [reply crates/code_generation/src/execution.rs:48]: The entire `PanicInfo` seems to implement `Display`?! can we use that?

**Review by schmidma — CHANGES_REQUESTED** [review]

**Comment by knoellle** [comment]
Closing in favor of #895

**Pull request closed without merging**